storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java

```diff
@@ -0,0 +1,518 @@
+            }
+            if (type.isInstance(o)) {
+                return;
+            }
+            throw new IllegalArgumentException("Field " + name + " must be of type " + type + ". Object: " + o + " actual type: " + o.getClass());
+        }
+    }
+
+    public static class StringValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object o) {
+            SimpleTypeValidator validator = new SimpleTypeValidator();
+            validator.validateField(name, String.class, o);
+        }
+    }
+
+    public static class BooleanValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object o) {
+            SimpleTypeValidator validator = new SimpleTypeValidator();
+            validator.validateField(name, Boolean.class, o);
+        }
+    }
+
+    public static class NumberValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object o) {
+            SimpleTypeValidator validator = new SimpleTypeValidator();
+            validator.validateField(name, Number.class, o);
+        }
+    }
+
+    public static class DoubleValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object o) {
+            SimpleTypeValidator validator = new SimpleTypeValidator();
+            validator.validateField(name, Double.class, o);
+        }
+    }
+
+    /**
+     * Validates a Integer.
+     */
+    public static class IntegerValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object o) {
+            validateInteger(name, o);
+        }
+
+        public void validateInteger(String name, Object o) {
+            if (o == null) {
+                return;
+            }
+            final long i;
+            if (o instanceof Number &&
+                    (i = ((Number) o).longValue()) == ((Number) o).doubleValue()) {
+                if (i <= Integer.MAX_VALUE && i >= Integer.MIN_VALUE) {
+                    return;
+                }
+            }
+            throw new IllegalArgumentException("Field " + name + " must be an Integer within type range.");
+        }
+    }
+
+    /**
+     * Validates a map of Strings to a map of Strings to a list.
+     * {str -> {str -> [str,str]}
+     */
+    public static class MapOfStringToMapValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object o) {
+            if (o == null) {
+                return;
+            }
+            ConfigValidationUtils.NestableFieldValidator validator = ConfigValidationUtils.mapFv(ConfigValidationUtils.fv(String.class, false),
+                    ConfigValidationUtils.mapFv(ConfigValidationUtils.fv(String.class, false),
+                            ConfigValidationUtils.listFv(String.class, false), false), true);
+            validator.validateField(name, o);
+        }
+    }
```

Comment:
Realizing this validator name was copied from before, it would be really nice to clean up the name here.  This really validates more than str->str->[str], but it also validates that the values of the nested map are lists of strings.  This is so specialized that maybe we want to rename the validator and its annotation to something like `ImpersonationAclValidator` and `@isImpersonationAcl`.

The reason is, should someone else suppose it is as generic as the name describes, they may find out it is being too strict and rejecting config entries that the javadoc and name do not say should be invalid.